hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieActiveTimeline.java

```diff
@@ -600,27 +617,37 @@ private void transitionState(HoodieInstant fromInstant, HoodieInstant toInstant,
   protected void transitionState(HoodieInstant fromInstant, HoodieInstant toInstant, Option<byte[]> data,
        boolean allowRedundantTransitions) {
     ValidationUtils.checkArgument(fromInstant.getTimestamp().equals(toInstant.getTimestamp()), String.format("%s and %s are not consistent when transition state.", fromInstant, toInstant));
+    String fromInstantFileName = instantFormat.getInstantFileName(fromInstant);
+    String toInstantFileName;
     try {
       if (metaClient.getTimelineLayoutVersion().isNullVersion()) {
         // Re-create the .inflight file by opening a new file and write the commit metadata in
-        createFileInMetaPath(fromInstant.getFileName(), data, allowRedundantTransitions);
-        Path fromInstantPath = getInstantFileNamePath(fromInstant.getFileName());
-        Path toInstantPath = getInstantFileNamePath(toInstant.getFileName());
+        createFileInMetaPath(fromInstantFileName, data, allowRedundantTransitions);
+        toInstantFileName = instantFormat.getInstantFileName(toInstant);
```

Comment:
Do we need to change this very old timeline layput specifically here? Should we handle it in the upgrade handler?